.editorconfig

```diff
@@ -323,3 +323,5 @@ dotnet_diagnostic.CA1805.severity = warning
 dotnet_diagnostic.CA1834.severity = warning
 # CA2000: Dispose objects before losing scope
 dotnet_diagnostic.CA2000.severity = warning
+# CA2208: Instantiate argument exceptions correctly
```

Comment:
Please include the reason you put in the PR description

> The culprit in this case is the second throw. CA2208 is telling us that the parameter that we pass in to ArgumentNullException should match a name of a parameter in the method signature. So in this case, the first throw does the correct thing. But not the second throw. A proper fix for this would be to not throw an ArgumentNullException, rather an ArgumentException. However, many of these instances are of a public method. So changing the exception being thrown is a breaking change.

Reply:
Good idea. Thanks